Backpack/Calendar/Classes/BPKCalendar.m

```diff
@@ -146,39 +152,54 @@ - (instancetype)initWithFrame:(CGRect)frame {
 - (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
                         maxDate:(BPKSimpleDate *)maxDate
                   configuration:(BPKCalendarConfiguration *)configuration
-         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
-    _configuration = configuration;
-
-    return [self initWithMinDate:minDate maxDate:maxDate selectionConfiguration:selectionConfiguration];
-}
-
-- (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
-                        maxDate:(BPKSimpleDate *)maxDate
-         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
+         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration
+                       calendar:(NSCalendar *)calendar
+                   dateProvider:(id<BPKMonthDateProvider>)dateProvider {
     BPKAssertMainThread();
-    if (_configuration == nil) {
-        _configuration = [BPKCalendarTrafficLightConfiguration new];
-    }
-
     self = [super initWithFrame:CGRectZero];
 
     if (self) {
-        self.selectionConfiguration = selectionConfiguration;
         self.minDate = minDate;
         self.maxDate = maxDate;
+        _configuration = configuration != nil ? configuration : [BPKCalendarTrafficLightConfiguration new];
+        self.selectionConfiguration = selectionConfiguration;
+        _gregorian = calendar;
+        _dateProvider = dateProvider;
         [self setup];
     }
 
     return self;
 }
 
+- (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
+                        maxDate:(BPKSimpleDate *)maxDate
+                  configuration:(BPKCalendarConfiguration *)configuration
+         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
+    NSCalendar *gregorian = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+    BPKCalendarMonthDateProvider *dateProvider = [[BPKCalendarMonthDateProvider alloc] initWithCalendar:gregorian];
+
+    return [self initWithMinDate:minDate
+                         maxDate:maxDate
+                   configuration:configuration
+          selectionConfiguration:selectionConfiguration
+                        calendar:gregorian
+                    dateProvider:dateProvider];
+}
+
+- (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
+                        maxDate:(BPKSimpleDate *)maxDate
+         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
+    BPKCalendarConfiguration *configuration = [BPKCalendarTrafficLightConfiguration new];
```

Comment:
Would we need the original if statement (!= nil on _configuration) to make sure we keep the functionality on par with the existing code?